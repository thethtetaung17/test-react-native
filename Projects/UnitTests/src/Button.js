import React, { Component } from 'react'
import { Text, View, TouchableOpacity } from 'react-native'

export default class Button extends Component {

    onPressHandler = () => {
        const { onPress, url } = this.props
        if (url) {
            Linking.openURL(url)
        }
        onPress()    
    }

    

    render() {
        const { buttonStyle, textStyle } = styles;
        const { label, primary } = this.props;
        const newButtonStyle = primary ? buttonStyle : [buttonStyle, { backgroundColor: '#f34541', borderBottomColor: '#a43532' }];
        return (
            <View>
                <Text> Button Component </Text>
                <TouchableOpacity onPress={this.onPressHandler} style={buttonStyle}>
                    <Text style={textStyle}>
                        {label}
                    </Text>
                </TouchableOpacity>
            </View>
        )
    }
}
Button.defaultProps = {
    primary: true,
}
const styles = {
    textStyle: {
        alignSelf: 'center',
        color: '#fff',
        fontSize: 16,
        fontWeight: '600',
    },
    buttonStyle: {
        height: 45,
        alignSelf: 'stretch',
        justifyContent: 'center',
        backgroundColor: '#38ba7d',
        borderBottomWidth: 6,
        borderBottomColor: '#1e6343',
        borderWidth: 1,
        marginLeft: 15,
        marginRight: 15
    }
};