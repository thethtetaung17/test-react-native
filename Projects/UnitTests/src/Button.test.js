import React from 'react'
import {shallow, configure} from 'enzyme'
import Button from './Button'
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()})
describe('Button', ()=> {
    describe('Renderig', () => {
        it('should match to snapshot', () => {
            const component = shallow(<Button/>)
            expect(component).toMatchSnapshot()
        });
    })
})