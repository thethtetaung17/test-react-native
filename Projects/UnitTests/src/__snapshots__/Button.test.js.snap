// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renderig should match to snapshot, Primary 1`] = `ShallowWrapper {}`;

exports[`Button Renderig should match to snapshot, Secondary 1`] = `ShallowWrapper {}`;
